src/dataset.ts

```diff
@@ -201,38 +215,41 @@ export class Savvy implements DataSet {
     }
     private cases : number;
     private data : Array<Row>;
-    private factors : Map<string, Map<number, string>>;
     private overflows : Map<string, Array<string>>;
     private fields : Map<string, Column>;
+    private _names : Map<string, string>;
     private _labels : Map<string, string>;
-    private _descriptions : Map<string, string>;
+    private _levels : Map<string, Map<number, string>>;
+    private _missings : Map<string, Set<number | string>>;
     /**
      *
      * @param parsed a {@link Parsed} object generated with a {@link SavParser}
      */
     constructor(parsed : Parsed) {
         this.cases = parsed.meta.cases;
+        this._names = parsed.internal.names;
         this._labels = parsed.internal.labels;
         this.data = parsed.rows;
-        this.factors = new Map(
+        this._levels = new Map(
             parsed.headers.map(header => [header.name, new Map()])
         );
-        parsed.internal.factors.forEach(
+        parsed.internal.levels.forEach(
             entry => entry.indices.forEach(
-                index => this.factors.set(
+                index => this._levels.set(
                     parsed.headers[index - 1].name,
                     entry.map
                 )
             )
         );
         this.fields = new Map();
         this.overflows = new Map();
-        this._descriptions = new Map();
+        this._labels = new Map();
```

Comment:
this._labels is set twice in the constructor. This will overwrite the above assignment, which I believe can be safely removed